scripts/transform.py

```diff
@@ -0,0 +1,219 @@
+import re
+
+import copy
+
+r"""
+This is the python script to generate c wrappers and corresponding
+headers for fortran (public) subroutines.
+
+It scans the files in src folder and generate the c wrappers in prg_c_interface.F90 and
+headers in prg_progress_mod.h.
+
+Note: the generated wrappers are not bug-free, it is only used to facilitate building
+c interrface.
+"""
+
+def get_public(fortran_code):
+
+   # Define the regex pattern to match lines starting with 'public ::'
+   pattern = re.compile(r'public ::(.*)')
```

Comment:
```suggestion
   pattern = re.compile(r'public *::(.*)')
```
The standard does not require exactly one whitespace after the public keyword.